packages/nimble-components/src/table-column/base/models/column-internals.ts

```diff
@@ -163,14 +174,27 @@ export class ColumnInternals<TColumnConfig> {
     @observable
     public currentSortDirection: TableColumnSortDirection = TableColumnSortDirection.none;
 
-    public constructor(options: ColumnInternalsOptions) {
+    public readonly validator: TColumnValidator;
+
+    public constructor(options: ColumnInternalsOptions<TColumnValidator>) {
         this.cellRecordFieldNames = options.cellRecordFieldNames;
         this.cellViewTemplate = createCellViewTemplate(options.cellViewTag);
         this.groupHeaderViewTemplate = createGroupHeaderViewTemplate(
             options.groupHeaderViewTag
         );
         this.delegatedEvents = options.delegatedEvents;
         this.sortOperation = options.sortOperation ?? TableColumnSortOperation.basic;
+        this.validator = options.validator;
+
```

Comment:
~~It's possible that the provided validator is initialized to be invalid I guess. Should we use its configured value?~~

```suggestion
this.validConfiguration = this.validator.isColumnValid;
```

~~Then we can remove the initialization since it would be set in constructor~~
```
@observable
public validConfiguration;
```
edit: not needed if removing prop in other comment